elfi/methods/results.py

```diff
@@ -203,6 +205,59 @@ def sample_means_array(self):
         """
         return np.array(list(self.sample_means.values()))
 
+    def save(self, fname=None, kind='csv'):
+        """Save samples in csv, json or pickle file formats.
+
+        Parameters
+        ----------
+        fname : str, required
+            File name to be saved
+        kind : str, optional
+            File format to be saved. Default format is csv
```

Comment:
Good to list exact supported strings here.

Consider inferring kind from fname?